pyfaidx/__init__.py

```diff
@@ -303,14 +303,25 @@ def __init__(self, filename, default_seq=None, key_function=lambda x: x.split()[
             else:
                 self._fasta_opener = bgzf.open
                 self._bgzf = True
-        elif filenameLower.endswith('.gz') or filenameLower.endswith('.bz2'):
-            raise RuntimeError("Compressed FASTA is only supported in BGZF "
-                               "format, via the BioPython bgzf module.")
+        elif filenameLower.endswith('.bz2') or filenameLower.endswith('.zip'):
+            raise RuntimeError(
+                "Compressed FASTA is only supported in BGZF format. Use "
+                "bgzip to compresss your FASTA.")
         else:
             self._fasta_opener = open
             self._bgzf = False
 
-        self.file = self._fasta_opener(filename, 'r+b' if mutable else 'rb')
+        try:
+            self.file = self._fasta_opener(filename, 'r+b' if mutable else 'rb')
+        except ValueError as e:
+            if str(e).find('BGZF') > -1:
+                raise RuntimeError(
+                    "Compressed FASTA is only supported in BGZF format. Use "
+                    "the samtools bgzip utility (instead of gzip) to "
+                    "compresss your FASTA.")
+            else:
+                raise
```

Comment:
OK, will do.  BTW, with your custom Extension subclasses, I don't think you need to have an `__init__` function. The `__init__` of `Exception` already allows you to pass `*args` and these are stored into the `args` attribute of the exception instance.  E.g.,

```python
e = Exception('hey', 3)
e.args
>>> ('hey', 3)
```

So I've made a class like this:

```python
class UnsupportedCompressionFormat(Exception):
    """
    Raised when a FASTA file is given with a recognized, but unsupported,
    compression extension.
    """
```

If you're ok with that, I can change the other classes too.  Another thing that could be done on testing those exceptions is to use `six.assertRaisesRegex` to check that the error message passed to create the exception instance is as expected. That can of course be done in another ticket, or not at all. It's a bit pedantic, but it guards against a test passing because the exception is raised but not for the reason you think it was raised.